x-pack/plugins/observability_solution/observability_onboarding/public/application/quickstart_flows/auto_detect/use_onboarding_flow.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import useInterval from 'react-use/lib/useInterval';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import useAsync from 'react-use/lib/useAsync';
+import { type AssetSOObject, type GetBulkAssetsResponse } from '@kbn/fleet-plugin/common';
+import { FETCH_STATUS, useFetcher } from '../../../hooks/use_fetcher';
+import { getOnboardingStatus } from './get_onboarding_status';
+import { getInstalledIntegrations } from './get_installed_integrations';
+import { type ObservabilityOnboardingContextValue } from '../../../plugin';
+
+export function useOnboardingFlow() {
+  const {
+    services: { fleet },
+  } = useKibana<ObservabilityOnboardingContextValue>();
+
+  // Create onboarding session
+  const { data, error, refetch } = useFetcher(
+    (callApi) =>
+      callApi('POST /internal/observability_onboarding/flow', {
+        params: {
+          body: {
+            name: 'auto-detect',
+          },
+        },
+      }),
+    [],
+    { showToastOnError: false }
+  );
+
+  const onboardingId = data?.onboardingFlow.id;
+
+  // Fetch onboarding progress
+  const {
+    data: progressData,
+    status: progressStatus,
+    refetch: refetchProgress,
+  } = useFetcher(
+    (callApi) => {
+      if (onboardingId) {
+        return callApi('GET /internal/observability_onboarding/flow/{onboardingId}/progress', {
+          params: { path: { onboardingId } },
+        });
+      }
+    },
+    [onboardingId]
+  );
+
+  const status = getOnboardingStatus(progressData);
+  const installedIntegrations = getInstalledIntegrations(progressData);
+
+  // Fetch metadata for installed Kibana assets
+  const assetsState = useAsync(async () => {
+    if (installedIntegrations.length === 0) {
+      return [];
+    }
+    const assetsMetadata = await fleet.hooks.epm.getBulkAssets({
+      assetIds: installedIntegrations
+        .map((integration) => integration.kibanaAssets)
+        .flat() as AssetSOObject[],
+    });
+    return installedIntegrations.map((integration) => {
+      return {
+        ...integration,
+        // Enrich installed Kibana assets with metadata from Fleet API (e.g. title, description, etc.)
+        kibanaAssets: integration.kibanaAssets.reduce<GetBulkAssetsResponse['items']>(
+          (acc, asset) => {
+            const assetWithMetadata = assetsMetadata.data?.items.find(({ id }) => id === asset.id);
+            if (assetWithMetadata) {
+              acc.push(assetWithMetadata);
+            }
+            return acc;
+          },
+          []
+        ),
+      };
+    });
+  }, [installedIntegrations.length]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Just a personal preference, no need for changes, but I'd keep metadata as a separate structure and let each installed integration look up stuff there while rendering.

Reply:
The drawback of that approach would be that you'd be making multiple unecessary HTTP requests for each integration separately to load that metadata in. Happy to refactor at a later stage if this causes issues but gonna keep as is for now